src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbol.cs

```diff
@@ -1391,6 +1391,62 @@ internal override void DecodeWellKnownAttribute(ref DecodeWellKnownAttributeArgu
                 // NullableAttribute should not be set explicitly.
                 arguments.Diagnostics.Add(ErrorCode.ERR_ExplicitNullableAttribute, arguments.AttributeSyntaxOpt.Location);
             }
+            else if (attribute.IsTargetAttribute(this, AttributeDescription.DisallowNullAttribute))
+            {
+                arguments.GetOrCreateData<PropertyWellKnownAttributeData>().HasDisallowNullAttribute = true;
```

Comment:
>arguments.GetOrCreateData<PropertyWellKnownAttributeData>().HasDisallowNullAttribute = true; [](start = 16, length = 92)

I would expect some kind of validation to report attributes that are not applicable to readonly/writeonly properties.

Reply:
Added some notes to the issue tracking warning for misused of attributes ([https://github.com/dotnet/roslyn/issues/36073](https://github.com/dotnet/roslyn/issues/36073))

---
In reply to: [295425342](https://github.com/dotnet/roslyn/pull/36444#discussion_r295425342) [](ancestors = 295425342)